src/meta/src/hummock/manager/mod.rs

```diff
@@ -996,7 +994,11 @@ where
             // policy.
             self.compactor_manager
                 .report_compact_task(context_id, compact_task);
-
+            // Tell compaction scheduler to resume compaction if there's any compactor becoming
+            // available.
+            if assigned_task_num == self.compactor_manager.max_concurrent_task_number() {
```

Comment:
assigned_task_num may also change due to task timeout and cancellation. So == may not work?

Reply:
As long as the task has an assignee_context_id and the original task number before report == the maximum number of tasks, we can determine that a compactor has become available.

Reply:
As long as the task has an assignee_context_id and the original task number before report == the maximum number of tasks, we can determine that a compactor has become available.

Reply:
It's also a bit hard for me to understand.

Reply:
`assigned_task_num` is only `Some` when `compact_task_assignment` contains the task being reported. Because `compact_task_assignment` is also used by `CompactorManager` to select the next idle compactor, "compactor availability change" is equal to "`assigned_task_num` is decreased from `max_concurrent_task_number()`". We needn't care about the `TaskStatus`, we only care if the task is already assigned.

Reply:
Got it. By the way there is a special task cancellation path: ,https://github.com/risingwavelabs/risingwave/blob/55875476eead2be30924bf679c73701419d36d64/src/meta/src/hummock/manager/context.rs#L52
I think it doesn't affect the correctness here.

Reply:
Also max_concurrent_task_number should be changed only through adding/removing compactor, which is true now.
For example if in the future we allow a compactor to change its max_concurrent_task_number online, we also need to  send a resume notification.

Reply:
Let me add a comment on `add_compactor` as a tip.